ReactAndroid/src/main/java/com/facebook/react/modules/websocket/WebSocketModule.java

```diff
@@ -162,30 +162,50 @@ public void onPong(Buffer buffer) {
 
       @Override
       public void onMessage(ResponseBody response) throws IOException {
-        String message;
-        try {
-          if (response.contentType() == WebSocket.BINARY) {
-            message = Base64.encodeToString(response.source().readByteArray(), Base64.NO_WRAP);
-          } else {
-            message = response.source().readUtf8();
+        WritableMap params = Arguments.createMap();
+        params.putInt("id", id);
+
+        if (mBlobsEnabled.containsKey(id) &&
+                mBlobsEnabled.get(id) &&
+                response.contentType() == WebSocket.BINARY) {
+          byte[] data;
+          try {
+            data = response.source().readByteArray();
+          } catch (IOException e) {
+            notifyWebSocketFailed(id, e.getMessage());
+            return;
           }
-        } catch (IOException e) {
-          notifyWebSocketFailed(id, e.getMessage());
-          return;
+          WritableMap blob = Arguments.createMap();
+          blob.putString("blobId", BlobModule.store(data));
+          blob.putInt("offset", 0);
+          blob.putInt("size", data.length);
+          params.putMap("data", blob);
+          params.putString("type", "blob");
+        } else {
+          String message;
+          try {
+            if (response.contentType() == WebSocket.BINARY) {
+              message = Base64.encodeToString(response.source().readByteArray(), Base64.NO_WRAP);
```

Comment:
This means data type can be "binary" but blobs can be disabled for this websocket so we pass all the data to JS as a huge Base64 string. Is there still any use case for this Base64 encoding of binary data once this PR lands? If we remove it it's a breaking change. But other than that, can blobs replace the API in all cases?

Why are blobs not automatically enabled when data type is "binary".